gtsam/slam/EssentialMatrixWithCalibrationFactor.h

```diff
@@ -0,0 +1,127 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file EssentialMatrixWithCalibrationFactor.h
+ *
+ * @brief A factor evaluating algebraic epipolar error with essential matrix and
+ * calibration as variables.
+ *
+ * @author Ayush Baid
+ * @author Akshay Krishnan
+ * @date April 23, 2021
+ */
+
+#pragma once
+
+#include <gtsam/geometry/EssentialMatrix.h>
+#include <gtsam/nonlinear/NonlinearFactor.h>
+
+#include <iostream>
+
+namespace gtsam {
+
+/**
+ * Factor that evaluates algebraic epipolar error (K^-1 p)'E (K^-1 p) for given
+ * essential matrix and calibration shared between two images.
+ */
+template <class CALIBRATION>
+class EssentialMatrixWithCalibrationFactor
```

Comment:
is there a simpler name here? Not sure that it conveys the overall gist of the factor.

`EssentialMatrixProjectionFactor`?

Reply:
This name is a bit tricky as we are not doing any reprojection to compute error, which is what this name suggests to me. 

The word projection does not hint that calibration is a parameter.